### PR TITLE
feat: add support for subscriptions

### PR DESCRIPTION
**Pull request by a-cordier**
see https://gravitee.atlassian.net/browse/GKO-359
see https://gravitee.atlassian.net/browse/GKO-360
see https://gravitee.atlassian.net/browse/GKO-361
see https://gravitee.atlassian.net/browse/GKO-362

depends on https://github.com/gravitee-io/gravitee-api-management/pull/9833

### CODE REVIEW

File: docs/guides/subscribe-to-jwt-plan/pki/private.key [hunk 1]
```diff
@@ -0,0 +1,51 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (graviteeio) [inline docs/guides/subscribe-to-jwt-plan/pki/private.key:1]: ### :warning: Aqua detected sensitive data in your code  
**Category:** AsymmetricPrivateKey   
**Description:** Asymmetric Private Key   
**Severity:** HIGH   
**Match:** ----BEGIN RSA PRIVATE KEY-----**************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************-----END RSA PRIVATE  
[This comment was created by Aqua Pipeline]

File: examples/apim/api_definition/v4/api-v4-with-jwt-plan.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    mode: DEFAULT
+    matchRequired: false
+  plans:
+    JWT:
+      name: "jwt"
+      description: "A JWT plan"
+      security:
+        type: "JWT"
+        configuration:
+          signature: "RSA_RS256"
```
Comment (kamiiiel) [inline examples/apim/api_definition/v4/api-v4-with-jwt-plan.yml:56]: Isn't it better if we use HMAC in our example and not rely on AM? WDYT?

File: examples/apim/subscription/subscription.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+kind: Subscription
+metadata:
+  name: api-key-subscription
+spec:
+  api:
+    name: api-v4-with-jwt-plan
+  application: 
+    name: simple-application-with-client-id
+  plan: JWT
+  endingAt: '2024-12-25T09:12:28Z'
```
Comment (kamiiiel) [inline examples/apim/subscription/subscription.yml:25]: Wouldn't be better if we add a later date so the tests will not fail?

File: internal/admission/subscription/validate.go [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+			sub.GetApiRef(), sub.GetPlan(),
+		)
+	}
+	return validatePlanSecurityType(plan, sub.GetPlan())
+}
+
+func validateApiState(api core.ApiDefinitionObject) *errors.AdmissionError {
+	if api.GetStatus() == nil || api.GetStatus().IsFailed() {
+		return errors.NewSeveref(
+			"unable to subscribe to API [%s] because it is in a a failed state", api.GetRef(),
```
Comment (kamiiiel) [inline internal/admission/subscription/validate.go:158]: one extra "a" in here

File: internal/admission/subscription/validate.go [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+			"unable to subscribe to API [%s] because it is not started", api.GetRef(),
+		)
+	}
+	return nil
+}
+
+func validateApplicationState(app core.ApplicationObject) *errors.AdmissionError {
+	if app.GetStatus() == nil || app.GetStatus().IsFailed() {
+		return errors.NewSeveref(
+			"unable to subscribe from application [%s] because it is in a a failed state", app.GetRef(),
```
Comment (kamiiiel) [inline internal/admission/subscription/validate.go:172]: one extra "a" in here too

File: api/model/subscription/subscription.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+	Plan string `json:"plan"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Format:=date-time
+	EndingAt *string `json:"endingAt,omitempty"`
+}
+
+type ApiRef struct {
+	refs.NamespacedName `json:",inline"`
+	// +kubebuilder:default:=ApiDefinition
+	Kind string `json:"kind,omitempty"`
```
Comment (kamiiiel) [inline api/model/subscription/subscription.go:43]: Do we really need this because we can find this ourselves when we retrieve the API, no?

**Pull request merged**